beacon-chain/rpc/beacon_chain_server.go

```diff
@@ -215,3 +305,35 @@ func (bs *BeaconChainServer) GetValidatorParticipation(
 ) (*ethpb.ValidatorParticipation, error) {
 	return nil, status.Error(codes.Unimplemented, "not implemented")
 }
+
+// startAndEndPage is a pagination heloer, it takes in the requested page token, size, total size,
+// and returns start, end page and the next page token.
+func (bs *BeaconChainServer) startAndEndPage(pageToken string, pageSize int, totalSize int) (int, int, string, error) {
```

Comment:
This doesn't need to be a pointer receiver as it is a pretty pure function. Can this go into some util like maybe a pagination.go file or shared/pagination?

Reply:
Good feedback. Done